src/effect-system/explicitInfer.ml

```diff
@@ -199,67 +175,36 @@ let stripHindleyMilnerLikeTy : Types.target_ty ->
 let instantiateVariable (x : variable) (scheme : Types.target_ty)
   : (Typed.expression * Types.target_ty * Typed.omega_ct list) =
   (* 1: Take the type signature apart *)
-  let skelVars, tyVarsWithSkels, dirtVars, tyCs, dirtCs, monotype =
+  let tyCs, dirtCs, monotype =
     (match stripHindleyMilnerLikeTy scheme with
-     | Some (a,b,c,d,e,f) -> (a,b,c,d,e,f)
+     | Some (a,b,c) -> (a,b,c)
      | None -> failwith "instantiateVariable: Non-HM type in the environment!") in
 
-  (* 2: Generate fresh skeleton, type, and dirt variables *)
-  let newSkelVars = List.map (fun _ -> CoreTypes.SkelParam.fresh ()) skelVars in
-  let newTyVars   = List.map (fun _ -> CoreTypes.TyParam.fresh ()) tyVarsWithSkels in
-  let newDirtVars = List.map (fun _ -> Types.fresh_dirt ()) dirtVars in
-
-  (* 3: Generate the freshening substitution *)
+  (* 2: Generate the freshening substitution *)
   let foldLeft f xs x0 = List.fold_left f x0 xs in (* GEORGE: Just for convenience *)
```

Comment:
I think no one uses function `foldLeft` anymore. If that is the case, it should be deleted.